### PR TITLE
Add redux state for batch creation requests, add request to create batch

### PR DESCRIPTION
**Pull request by nickgraz**
- Using the `createAsyncThunk` redux utility

### CODE REVIEW

**Comment by nickgraz** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1884** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1884?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #1885** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1885?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1886** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1886?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/1884?utm_source=stack-comment).

File: src/redux/features/batches/batchesAsyncThunks.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+export const requestSaveBatch = createAsyncThunk(
+  'batches/save',
+  async (request: { batch: SavableBatch; timezone: string }, { rejectWithValue }) => {
+    const { batch, timezone } = request;
+
+    let response: (Response & BatchData) | (Response & BatchId) | undefined;
+    let responseQuantities: Partial<Response> = { requestSucceeded: true, error: undefined };
+
+    if (batch.id === -1) {
```
Comment (charlie-tf) [inline src/redux/features/batches/batchesAsyncThunks.ts:20]: Should this be `if (batch && batch.id === -1)` or `if (batch?.id === -1)`?

**Review by charlie-tf — COMMENTED** [review]
Just one question

File: src/redux/features/batches/batchesAsyncThunks.ts (hunk 1 above)
Comment (nickgraz) [reply src/redux/features/batches/batchesAsyncThunks.ts:20]: By this point we have already guaranteed that the batch exists, we've narrowed it to `SavableBatch`. In other words, you can't call this request without a batch

**Review by charlie-tf — APPROVED** [review]

**Comment by nickgraz** [comment]
### Merge activity

* **Dec 11, 6:23 PM**: @@nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/1884).

**Pull request merged**